### ISSUE TITLE
avoid_positional_boolean_parameters: false positive on setter with @visibleForTesting

### ISSUE THREAD

**Issue by lukepighetti** [dart-lang/linter#3922]
**Describe the issue**
False positive on boolean setters

**To Reproduce**

```dart
class Foo {
  bool _value = false;
  @visibleForTesting
  set foo(bool value) => _value = value; // lint
}
```

**Expected behavior**

```dart
class Foo {
  bool _value = false;
  @visibleForTesting
  set foo(bool value) => _value = value; // no lint
}
```

**Additional context**

**Comment by pq** [comment]
Good catch.  Thanks @lukepighetti!

**Comment by SkinnyMind** [comment]
Also have false positive on class arguments:

```dart
class Foo {
  const Foo({required this.callback});

  final void Function(bool) callback; // lint
}
```

**Comment by SkinnyMind** [comment]
I realized it's probably not a false positive, since in 1.32.0 addition of "handle typedefs". I have a question then, previously code like i posted above worked in Flutter by passing callbacks to, e.g. `onPressed` arguments of widgets:
```dart
class CustomChip extends StatelessWidget {
  const CustomChip({required this.onSelected});

  final void Function(bool) onSelected;

  @override
  Widget build(BuildContext context) {
    return RawChip(
     // ...
      onSelected: onSelected,
    );
  }
}
```

But in 1.32.0 line with `final void Function(bool) onSelected;` is being highlighted. It can be fixed by changing it to named parameter `final void Function({bool value}) onSelected;`

However after that change line `onSelected: onSelected,` is giving error "The argument type 'void Function({bool value})' can't be assigned to the parameter type 'void Function(bool)?'".

Changing parameter in function to named also leads to error in places where the `CustomChip` is referenced:
```dart
CustomChip(onSelected: (value) {}) // lint "The argument type 'void Function(bool)' can't be assigned to the parameter type 'void Function({bool value})'"
```

**Comment by srawlins** [comment]
I cannot reproduce this, and there is now a test case asserting that it does not report setters. I'll close for now, but please update the issue if you can reproduce.

@SkinnyMind Please open a separate issue for your concern. Sorry for the extreme delay! I think the issue you raise is valid though.

**Comment by SkinnyMind** [comment]
@srawlins Error that i was having can be fixed by adding `required` keyword in both declaration and reference of callback parameter. E.g.:
```diff
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
-          child: CustomChip(onSelected: (value) => print(value)), // Error: The argument type 'void Function(dynamic)' can't be assigned to the parameter type 'void Function({bool value})'.
+          child: CustomChip(onSelected: ({required value}) => print(value)), // No error
        ),
      ),
    );
  }
}

class CustomChip extends StatelessWidget {
  const CustomChip({required this.onSelected, super.key});

- final void Function({bool value}) onSelected;
+ final void Function({required bool value}) onSelected;

  @override
  Widget build(BuildContext context) {
    return RawChip(
      label: const Text('Text'),
      onSelected: (value) => onSelected(value: value),
    );
  }
}

```

I'm not sure if it's the correct way of doing it, but everything works and makes sense if you think about declaration of non-nullable named parameters.
